src/robot/SessionManager.test.ts

```diff
@@ -26,7 +26,7 @@ const mockHealthyHeartbeat = (_req, _md, cb) => {
 describe('SessionManager', () => {
   beforeEach(() => {
     sm = new SessionManager(host, transport);
-    vi.mock('./gen/robot/v1/robot_pb_service.esm');
+    vi.mock('./gen/robot/v1/robot_pb_service');
```

Comment:
Not a change introduced by this PR, but `vi.mock` gets hoisted to the top of the file and runs only once. It should probably be moved out of the `beforeEach` to make that clear

(You could also replace it with `vi.doMock`, which will not be hoisted and run on every `beforeEach`, but due to vitest's module caching, this won't do anything other than potentially slow the test suite down)